server/embed/etcd.go

```diff
@@ -388,6 +399,24 @@ func (e *Etcd) Config() Config {
 // Close gracefully shuts down all servers/listeners.
 // Client requests will be terminated with request timeout.
 // After timeout, enforce remaning requests be closed immediately.
+//
+// The rough workflow to shut down etcd:
+//  1. close the `stopc` channel, so that all error handlers (child
+//     goroutines) won't send back any errors anymore;
+//  2. close all client and metrics listeners, so that etcd server
```

Comment:
Do we expect closing clients, listiners, servers is blocking? How we know we can immediately proceed?

Reply:
Closing listeners is only to stop receiving new connections, it won't close the already established connection. We should gracefully shutdown the etcdserver.

Reply:
Reverted this change, to reduce the review burden.